uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-files-tab.js

```diff
@@ -84,11 +91,76 @@ define([
       return !attachment.attachment_id || vm.canManage;
     };
 
+    /**
+     * Gets called when the component is destroyed
+     */
+    vm.$onDestroy = function () {
+      // destroy all the event
+      events.map(function (event) {
+        event();
+      });
+    };
+
     (function init () {
-      vm.supportedFileTypes = _.keys(sharedSettings.fileUploader.allowedMimeTypes);
-      vm.request.loadAttachments();
+      $q.all([
+        loadSupportedFileTypes(),
+        vm.request.loadAttachments()
+      ])
+      .then(function () {
+        $rootScope.$broadcast('Request Updated');
+      })
+      .finally(function () {
+        vm.filesLoaded = true;
+      });
+      events.push($rootScope.$on('uploadFiles: start', uploadAttachments));
     }());
 
+    /**
+     * Load file extensions which are supported for upload and creates uploader object
+     *
+     * @return {Promise}
+     */
+    function loadSupportedFileTypes () {
+      var allowedMimeTypes;
+
+      return OptionGroup.valuesOf('safe_file_extension')
+        .then(function (extensions) {
+          allowedMimeTypes = {};
+
+          vm.supportedFileTypes = extensions.map(function (ext) {
```

Comment:
you are mixing two things here in the same `map`

1) You are creating the `vm.supportedFileTypes`
2) You are populating the `allowedMimeTypes`

The fact that both are happening the in the `map` that you use to define `vm.supportedFileTypes` is very confusing. Please do the two thing separately